File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;
map<string, int> s;
string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    for(string p:completion) {
        s[p]++;
    }
    for(string p:participant) {
        s[p]--;
    }
    for(auto i:s){
        if(i.second<0)return i.first;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
	public String solution(String[] participant, String[] completion) {
		String answer = "";

		Map<String, Integer> map = new HashMap<>();

		for (String s : participant) {
			int a = map.getOrDefault(s, 0);
			map.put(s, a + 1);
		}

		for (String s : completion) {
			int a = map.get(s);
			map.put(s, a - 1);
		}

		
		for (var iter : map.entrySet()) {
			if (iter.getValue() != 0) {
				return iter.getKey();
			}
		}

		return answer;
	}
}